lib/SILOptimizer/Mandatory/MoveOnlyChecker.cpp

```diff
@@ -543,6 +543,46 @@ bool MoveOnlyChecker::searchForCandidateMarkMustChecks() {
       if (!mmci || !mmci->hasMoveCheckerKind())
         continue;
 
+      // Handle guaranteed/owned move only typed arguments.
+      //
+      // We are pattern matching against these patterns:
+      //
+      // bb0(%0 : @guaranteed $T):
+      //   %1 = copy_value %0
```

Comment:
@gottesmm if an argument is marked `@moveOnly`, then we need to be sure no `copy_value` exists in the argument, at least by the time this diagnostic runs. This is true for both owned and guaranteed.

Reply:
I talked with Andy about this, what actually happens here is that as part of the diagnostic I eliminate the copy_value and the mark_must_check. I did it this way since today copy propagation doesn't support guaranteed parameters (although we could extend it, I decided to just hack around it).

Reply:
The right thing here is I think to just see if we can get copy propagation working on guaranteed things and rip this out.

Reply:
@gottesmm for future reference, copy propagation always handled guaranteed function arguments. It's done in `RewriteInnerBorrowUses` and `CanonicalizeBorrowScope::visitBorrowScopeUses`.